php/media/class-gallery.php

```diff
@@ -0,0 +1,321 @@
+			}
+
+			unset( $input[ $key ] );
+		}
+
+		return $input;
+	}
+
+	/**
+	 * Returns JSON format of the current config.
+	 *
+	 * @return string|null
+	 */
+	public function get_json() {
+		return count( $this->get_config() ) ? wp_json_encode( $this->get_config() ) : null;
+	}
+
+	/**
+	 * Register frontend assets for the gallery.
+	 */
+	public function enqueue_gallery_library() {
+		wp_enqueue_script(
+			self::GALLERY_LIBRARY_HANDLE,
+			self::GALLERY_LIBRARY_URL,
+			array(),
+			$this->media->plugin->version,
+			true
+		);
+
+		$config = $this->get_config();
+
+		if ( $this->woocommerce_active() ) {
+			$product = wc_get_product();
+
+			if ( $product ) {
+				$config['mediaAssets'] = $this->get_image_data( $product->get_gallery_image_ids() );
+			}
+		}
+
+		wp_localize_script(
+			self::GALLERY_LIBRARY_HANDLE,
+			'cloudinaryGallery',
+			array( 'config' => wp_json_encode( $config ) )
+		);
+
+		wp_enqueue_script(
+			'cloudinary-gallery-init',
+			$this->media->plugin->dir_url . 'assets/dist/gallery-init.js',
+			array( self::GALLERY_LIBRARY_HANDLE ),
+			$this->media->plugin->version,
+			true
+		);
+	}
+
+	/**
+	 * Register blocked editor assets for the gallery.
+	 */
+	public function block_editor_scripts_styles() {
+		$this->enqueue_gallery_library();
+
+		wp_enqueue_style(
+			'cloudinary-gallery-block-css',
+			$this->media->plugin->dir_url . 'assets/dist/block-gallery.css',
+			array(),
+			$this->media->plugin->version
+		);
+
+		wp_enqueue_script(
+			'cloudinary-gallery-block-js',
+			$this->media->plugin->dir_url . 'assets/dist/block-gallery.js',
+			array( 'wp-blocks', 'wp-editor', 'wp-element', self::GALLERY_LIBRARY_HANDLE ),
+			$this->media->plugin->version,
+			true
+		);
+
+		wp_localize_script(
+			'cloudinary-gallery-block-js',
+			'cloudinaryGalleryApi',
+			array(
+				'endpoint' => rest_url( REST_API::BASE . '/image_data' ),
+				'nonce'    => wp_create_nonce( 'wp_rest' ),
+			)
+		);
+	}
+
+	/**
+	 * Check if WooCommerce is active.
+	 *
+	 * @return bool
+	 */
+	protected function woocommerce_active() {
+		return class_exists( 'WooCommerce' ) && function_exists( 'wc_get_product' );
+	}
+
+	/**
+	 * Checks if the Cloudinary Gallery Widget is enabled.
+	 *
+	 * @return bool
+	 */
+	public function gallery_enabled() {
+		return isset( $this->original_config['enable_gallery'] ) && 'on' === $this->original_config['enable_gallery'];
+	}
+
+	/**
+	 * Fetches image public id and transformations.
+	 *
+	 * @param array|int[]|array[] $images An array of image IDs or a multi-dimensional array with url and id keys.
+	 *
+	 * @return array
+	 */
+	public function get_image_data( array $images ) {
+		$image_data = array();
+
+		foreach ( $images as $index => $image ) {
+			$image_id  = is_int( $image ) ? $image : $image['id'];
+			$image_url = is_int( $image ) ? $this->media->cloudinary_url( $image_id ) : $image['url'];
+
+			if ( ! $this->media->sync->is_synced( $image_id ) ) {
+				continue;
+			}
```

Comment:
@dugajean - This should be the first check done.

Reply:
Good call 👍